### PR TITLE
Add samples for DES-ECB, wrap with template and unwrap with template

### PR DESCRIPTION
**Pull request by sushrb**
*Description of changes:*

The samples and files added in this pull request include:
- DES_ECB (src/encrypt/des_ecb.c)
- WRAP_TEMPLATE (src/wrapping/wrap_with_template.c)
- UNWRAP_WITH_TEMPLATE (src/wrapping/unwrap_with_template.c)

### CODE REVIEW

File: src/encrypt/des_ecb.c [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    // Prepare the mechanism
+    CK_MECHANISM mech = {CKM_DES3_ECB, NULL, 0};
+
+    //**********************************************************************************************
+    // Encrypt
+    //**********************************************************************************************
+
+    rv = funcs->C_EncryptInit(session, &mech, des_key);
+    if (CKR_OK != rv) {
+        printf("Encryption Init failed: %lu\n", rv);
```
Comment (rday) [inline src/encrypt/des_ecb.c]: Could you update the failure cases to be `fprintf(stderr, ...)`?

File: src/encrypt/des_ecb.c [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    return rv;
+}
+
+int main(int argc, char **argv) {
+    CK_RV rv;
+    CK_SESSION_HANDLE session;
+
+    struct pkcs_arguments args = {};
+    if (get_pkcs_args(argc, argv, &args) < 0) {
+        return 1;
```
Comment (rday) [inline src/encrypt/des_ecb.c]: And the returns to be `EXIT_FAILURE` or `EXIT_SUCCESS`?

File: README.md [hunk 3]
```diff
@@ -28,7 +28,7 @@ project. Run make to build the examples.
 ```
 mkdir build/
 cd build/
-cmake .. -DHSM_USER=<user> -DHSM_PASSWORD=<password>
+cmake .. -DHSM_USER=<user> -DHSM_PASSWORD=<password> -DTRUSTED_WRAPPING_KEY_HANDLE=<trusted_key>
```
Comment (rday) [inline README.md]: Is this needed? Looks like the wrapping key is a CLI argument.

File: src/encrypt/des_ecb.c (hunk 1 above)
Comment (sushrb) [reply src/encrypt/des_ecb.c]: resolved.

File: src/encrypt/des_ecb.c (hunk 2 above)
Comment (sushrb) [reply src/encrypt/des_ecb.c]: resolved.

**Pull request merged**